### PR TITLE
[MB-15883, MB-15796] [FE] Create/reuse modal to edit the SIT address for the SIT service item

### PR DESCRIPTION
**Pull request by afoon**
## Summary

This PR implements the frontend for [MB-15797 - TOO can create/set SIT address change](https://dp3.atlassian.net/browse/MB-15797)

## Verification Steps for the Author

These are to be checked by the author.

- [ ] Tested in the Experimental environment (for changes to containers, app startup, or connection to data stores)
- [ ] Have the Jira acceptance criteria been met for this change?

## Verification Steps for Reviewers
- [ ] Clicking on “Edit” link in the “Approved service item” section will display a modal to Edit SIT delivery address
- [ ] Matches the [designs](https://www.figma.com/file/5AhuEkaH3bJVrtffU3wp22/Milmove-Office-App?type=design&node-id=11287-118005&t=o7cKNmt0kH2rxxWY-0) , particularly: 
  - [ ] HHG blue label is displayed at top left corner
  - [ ] Title of modal is Edit service item
   - [ ] SIT delivery address in a white section
   - [ ] Initial SIT delivery address
   - [ ] Address Form
       - [ ] Distance warning
       - [ ] Address 1
       - [ ] Address 2 (optional)
       - [ ] City
       - [ ] State
       - [ ] ZIP
  - [ ] Office remarks
  - [ ] Save button
  - [ ] Cancel button
  - [ ] Grayed out if address and office remarks are empty
  - [ ] Closes the modal when cancel, x, or background is clicked
 - [ ] Has unit test

### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/run-storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/about/application-setup/milmove-local-client/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/about/development/testing)

### How to test

1. `For local env only`: After running the server, run `DEVSEED_SUBSCENARIO=sit_address_updates make db_dev_e2e_populate
` in the terminal
2. Access the Office App
3. Login as a TOO
4. Navigate to `SITUP1` or any move starting with `SITUP`
5.  Click Move Task Order 
6. Scroll down to the services Items
7. Click Edit on the Domestic SIT item
    - [ ] Clicking on “Edit” link in the “Approved service item” section will display a modal to Edit SIT delivery address
9. Change the Address, including the Zip Code to one of the following: `80011`, `78705` or `85309`.
10. Add remarks
11. Click save

### Frontend

- [ ] There are no aXe warnings for UI.
- [ ] This works in [Supported Browsers and their phone views](https://transcom.github.io/mymove-docs/docs/about/supported-browsers) (Chrome, Firefox, Edge).
- [ ] There are no new console errors in the browser devtools.
- [ ] There are no new console errors in the test output.
- [ ] If this PR adds a new component to Storybook, it ensures the component is fully responsive, OR if it is intentionally not, a wrapping div using the `officeApp` class or custom `min-width` styling is used to hide any states the would not be visible to the user.

## Screenshots
| Type | Screenshot |
|--|--|
| Modal with Edit Sit Address Form | Screenshot 2023-05-23 at 4 26 15 PM|
| Base Modal | Screenshot 2023-05-23 at 4 27 42 PM|

### CODE REVIEW

File: src/components/Office/ServiceItemUpdateModal/ServiceItemUpdateModal.jsx [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+  onSave,
+  closeModal,
+  title,
+  serviceItem,
+  children: content,
+  initialValues,
+  validations,
+}) => {
+  const serviceItemUpdateModalSchema = Yup.object().shape({
+    officeRemarks: Yup.string().required('Required'),
```
Comment (afoon) [inline src/components/Office/ServiceItemUpdateModal/ServiceItemUpdateModal.jsx:57]: Usually this is nullable but the backend requires the remarks to complete the request. If  we want TOOs to be able to leave this blank, a change will need to be made there.

**Review by AnnaGingle — CHANGES_REQUESTED** [review]
- Address formatted incorrectly
- Office Remarks field should be 136px in height
- Missing border & additional padding around Final SIT delivery content

Anna's private board - 10648

**Comment by rogeruiz** [comment]
thanks @AnnaGingle I'll take a look at this feedback and have a new version for you to review shortly.

**Review by AnnaGingle — APPROVED** [review]
Great work!

File: src/utils/shipmentDisplay.jsx [hunk 2]
```diff
@@ -9,11 +9,11 @@ import affiliations from 'content/serviceMemberAgencies';
...
-    <>
-      {streetAddress1 && <>{streetAddress1},&nbsp;</>}
-      {streetAddress2 && <>{streetAddress2},&nbsp;</>}
-      {city ? `${city}, ${state} ${postalCode}` : postalCode}
-    </>
+    <address>
+      <span>{streetAddress1 && <>{streetAddress1},&nbsp;</>}</span>
+      <span>{streetAddress2 && <>{streetAddress2},&nbsp;</>}</span>
+      <span>{city ? `${city}, ${state} ${postalCode}` : postalCode}</span>
+    </address>
```
Comment (afoon) [inline src/utils/shipmentDisplay.jsx:16]: @rogeruiz This style change is affecting a few other places that also use formatAddress. It's breaking some test and happo renderings. @AnnaGingle can you confirm the changes in spacing seen in the Happo renderings are fine? I've added an example of on the visual changes.
Screenshot 2023-05-24 at 3 56 55 PM

**Review by rogeruiz — DISMISSED** [review]
lgtm 🌈

File: src/utils/shipmentDisplay.jsx (hunk 2 above)
Comment (rogeruiz) [reply src/utils/shipmentDisplay.jsx:16]: @AnnaGingle I'm going to update this PR to make this not a breaking visual change across the other components that @afoon mentioned here.

Comment (rogeruiz) [reply src/utils/shipmentDisplay.jsx:34]: This was fixed in https://github.com/transcom/mymove/pull/10648/commits/68021b0905ada00fa620b1aca45d7f823037ab82

**Review by rogeruiz — APPROVED** [review]
lgtm 🌈

**Comment by donaldthai** [comment]
@Mergifyio update

**Pull request merged**